src/extension-provider/createExternalExtensionProvider.ts

```diff
@@ -35,13 +37,9 @@ export function createExternalExtensionProvider() {
   return provider;
 }
 
-function getMetaMaskId() {
-  switch (browser?.name) {
-    case 'chrome':
-      return config.CHROME_ID;
-    case 'firefox':
-      return config.FIREFOX_ID;
-    default:
-      return config.CHROME_ID;
-  }
+function getExtensionId(typeOrId: ExtensionType) {
+  const ids =
+    browser?.name === 'firefox' ? config.firefoxIds : config.chromeIds;
+  // @ts-expect-error Ignoring indexing error since we have a fallback value
+  return ids[typeOrId] ?? typeOrId;
```

Comment:
Would prefer some kind of type cast here over ignoring TS errors for the entire line.
```suggestion
  return ids[typeOrId as keyof typeof ids] ?? typeOrId;
```

Reply:
Good point!